appinventor/blocklyeditor/src/replmgr.js

```diff
@@ -333,14 +334,17 @@ Blockly.ReplMgr.putYail = (function() {
                 return;
             }
 
-            // Add the protect-enum macro (used by dropdown blocks).
-            code += "(define-syntax protect-enum " +
-                "  (lambda (x) "+
-                "    (syntax-case x () " +
-                "      ((_ enum-value number-value) " +
-                "        (if (< com.google.appinventor.components.common.YaVersion:BLOCKS_LANGUAGE_VERSION 33) " +
-                "          #'number-value " +
-                "          #'enum-value)))))";
+            if (!sentMacros) {
+                // Add the protect-enum macro (used by dropdown blocks).
+                code = "(define-syntax protect-enum " +
```

Comment:
Did you mean to change this from += to =?

Reply:
Yes, I made this a prepend rather than append so that the syntax gets defined first, otherwise it can be referenced in the code before it's defined, which leads to bad times.